### PR TITLE
feat(components): [badge] add show-zero prop

### PR DESCRIPTION
**Pull request by IceyWu**
Add show-zero prop,whether to show badge when valueis zero

### CODE REVIEW

File: packages/components/badge/src/badge.vue [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <slot />
     <transition :name="`${ns.namespace.value}-zoom-in-center`">
       <sup
-        v-show="!hidden && (content || isDot)"
+        v-show="isShowBadge"
```
Comment (kooriookami) [inline packages/components/badge/src/badge.vue:6]:
```suggestion
        v-show="!hidden && (content || isDot)"
```
Please keep it, I will explain later.

File: packages/components/badge/src/badge.vue [hunk 2]
```diff
@@ -39,6 +39,11 @@ const content = computed<string>(() => {
   return `${props.value}`
 })
 
+const isShowBadge = computed(() => {
+  const isShowVal = +content.value === 0 ? props.showZero : content.value
+  return !props.hidden && (isShowVal || props.isDot)
+})
```
Comment (kooriookami) [inline packages/components/badge/src/badge.vue:45]:
```suggestion
```
You can't tell the `content.value` is a string or number type, please move the judgment to the computed `content`.

```
const content = computed<string>(() => {
  if (props.isDot) return ''

  if (props.value === 0 && !props.showZero) return ''

  if (isNumber(props.value) && isNumber(props.max)) {
    return props.max < props.value ? `${props.max}+` : `${props.value}`
  }
  return `${props.value}`
})
```

**Review by kooriookami — CHANGES_REQUESTED** [review]

File: docs/en-US/component/badge.md [hunk 3]
```diff
@@ -58,6 +58,7 @@ badge/dot
 | is-dot | if a little dot is displayed.                                                 | ^[boolean]                                                         | false   |
 | hidden | hidden badge.                                                                 | ^[boolean]                                                         | false   |
 | type   | badge type.                                                                   | ^[enum]`'primary' \| 'success' \| 'warning' \| 'danger' \| 'info'` | danger  |
+| show-zero | Whether to show badge when value is zero.                                  | ^[boolean]                                                         | true    |
```
Comment (kooriookami) [inline docs/en-US/component/badge.md:61]:
```suggestion
| show-zero  ^(2.6.0) | Whether to show badge when value is zero.                                  | ^[boolean]                                                         | true    |
```
Please add a version tag.

**Review by kooriookami — CHANGES_REQUESTED** [review]

File: docs/en-US/component/badge.md (hunk 3 above)
Comment (IceyWu) [reply docs/en-US/component/badge.md:61]: ok

**Review by kooriookami — APPROVED** [review]

**Review by btea — CHANGES_REQUESTED** [review]
@IceyWu Could you add a test case ?

**Comment by IceyWu** [comment]
> @IceyWu Could you add a test case ?

OK

File: packages/components/badge/__tests__/badge.test.tsx [hunk 4]
```diff
@@ -62,4 +62,15 @@ describe('Badge', () => {
...
+
+  test('showZero', async () => {
+    const badgeValue = ref(0)
+    const wrapper = mount(() => (
+      <Badge showZero={true} value={badgeValue.value} />
+    ))
+    expect(wrapper.find('.el-badge__content').text()).toEqual('0')
+    badgeValue.value = 80
+    await nextTick()
+    expect(wrapper.find('.el-badge__content').text()).toEqual('80')
```
Comment (kooriookami) [inline packages/components/badge/__tests__/badge.test.tsx:74]: When changing `showZero` to false, it's necessary to test the value `0`.

**Review by kooriookami — CHANGES_REQUESTED** [review]

File: packages/components/badge/__tests__/badge.test.tsx (hunk 4 above)
Comment (IceyWu) [reply packages/components/badge/__tests__/badge.test.tsx:74]: Should I add the situation where showZero is false？

Comment (kooriookami) [reply packages/components/badge/__tests__/badge.test.tsx:74]: Yes, set `const showZero= ref(true);` and change the vaule.

**Review by kooriookami — APPROVED** [review]

File: packages/components/badge/src/badge.vue [hunk 5]
```diff
@@ -33,6 +33,8 @@ const ns = useNamespace('badge')
 const content = computed<string>(() => {
   if (props.isDot) return ''
 
+  if (props.value === 0 && !props.showZero) return ''
```
Comment (btea) [inline packages/components/badge/src/badge.vue:36]: The judgment here should be placed on line 39.
```ts
if (props.max < props.value) {
  return `${props.max}+`
}
return props.value === 0 && !props.showZero ? '' : `${props.value}`
```
For example, when `props.value` is 0, `showZero` is true, but `props.max` is `-1`, the badge should be displayed.

Comment (IceyWu) [reply packages/components/badge/src/badge.vue:36]: Like this? 
```suggestion
 const content = computed<string>(() => {
  if (props.isDot) return ''
  if (isNumber(props.value) && isNumber(props.max)) {
    if (props.max < props.value) {
      return `${props.max}+`
    }
    return props.value === 0 && !props.showZero ? '' : `${props.value}`
  }
  return `${props.value}`
})
```

Comment (IceyWu) [reply packages/components/badge/src/badge.vue:36]: Ok

**Review by btea — APPROVED** [review]

**Pull request merged**